main.go

```diff
@@ -112,25 +113,29 @@ func main() {
 		fetchCtx = ctx
 	}
 
-	// set up default middleware stack + timer
-	var (
-		stack        []middleware.MessageHandlerDecorator
-		cancelDelete context.CancelFunc
-	)
+	// set up middleware stack with timing metrics
+	var stack []middleware.MessageHandlerDecorator
 	{
-		ctx, cancel := context.WithCancel(context.Background())
-		stack = middleware.DefaultStack(ctx, sqsSvc)
-		cancelDelete = cancel
-
-		// track job execution time
 		jobTime := dd.NewTiming("job.time", 1).With("version", Version)
-		stack = append(stack, TrackJobTime(jobTime))
+		timing := TrackJobTime(jobTime)
+
+		messageAge := dd.NewGauge("job.age").With("version", Version)
+		ageTracker := middleware.TrackMessageAge(time.Second, func(age float64) {
+			messageAge.Set(age)
+		})
+
+		stack = []middleware.MessageHandlerDecorator{ageTracker, timing}
 	}
 
 	h := middleware.ApplyDecoratorsToHandler(w.HandleMessage, stack...)
 	c := sqsconsumer.NewConsumer(sqsSvc, h)
+	c.ExtendVisibilityTimeoutBySeconds = 600
+	c.ExtendVisibilityTimeoutEvery = 7 * time.Minute
+	c.ReceiveVisibilityTimoutSeconds = 600
```

Comment:
There are a lot of magic durations in here. They should probably have Env overrides.